include/mata/nfa/plumbing.hh

```diff
@@ -44,8 +46,8 @@ inline void complement(
         Nfa*               result,
         const Nfa&         aut,
         const Alphabet&    alphabet,
-        const StringMap&  params = {{"algorithm", "classical"},
-                                    {"minimize", "false"}}) { *result = complement(aut, alphabet, params);
+        const ParameterMap&  params = {{ "algorithm", "classical"},
```

Comment:
Is ParameterMap to be used in dispatch only?
What about this other map that could be used for the optional parameters?
If it is something different, then maybe some names that distinguish the two would be good.

Reply:
> Is ParameterMap to be used in dispatch only?

No.

>  What about this other map that could be used for the optional parameters?
What is `this other map`? There is only one parameter map (`ParameterMap`). There is also `StateRenaming`, if that is what you are talking about, but that is a completely different map with different purposes.

Reply:
By "this other map" I meant a parameter map for optional parameters such as renaming of states (that you called output parameters below I think).

There is some confusion coming from that we are mixing discussion about these switches in the dispatch functions, like algorithm=classical, that define what the function actually does, 
and optional parameters such as state renaming (state renaming is one parameter within a potential parameter map storing optional parameters), that do not really influence the main functionality byt do something really optional on top of that, like store some side information.

Actually, I would rather talk about this since there is a lot of misunderstanding. Too time consuming to solve it here.

Reply:
We decided to keep the parameter map as it is now. Might be modified in the future if we find the current solution not sufficient for our use cases.